weld-rendering/app/src/main/java/styler/weld/rendering/viewholder/ItemListViewHolder.kt

```diff
@@ -6,25 +6,41 @@ import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
 import styler.weld.rendering.R
 import styler.weld.rendering.adapter.HorizontalAdapter
+import styler.weld.rendering.models.local.itemlist.ItemListItem
 import styler.weld.rendering.models.remote.WidgetData
 
 class ItemListViewHolder(
-    itemView: View,
-    private val widgetDataList: List<WidgetData>?
+    itemView: View
 ) : BaseViewHolder(itemView) {
     private val title: TextView = itemView.findViewById(R.id.text_view_title_item)
     private val recyclerView: RecyclerView = itemView.findViewById(R.id.recycler_view_item)
-    override fun bindData() {
-        widgetDataList?.forEach { widgetData ->
-            if (widgetData.type == "item_list") {
-                val data = widgetData.data
-                recyclerView.apply {
-                    adapter = HorizontalAdapter(data.items, "item")
-                    layoutManager =
-                        LinearLayoutManager(itemView.context, LinearLayoutManager.HORIZONTAL, false)
-                    title.text = widgetData.title
-                }
+    override fun bindData(data: WidgetData) {
+        recyclerView.apply {
+            // TODO: Left for future reference, but can probably be deleted
+//            val itemList = data.data["items"] as List<BaseItem>
+//            val itemList = (data.data["items"] as List<*>).filterIsInstance<ItemListItem>().takeIf {
+//                it.size == (data.data["items"] as List<*>).size
+//            }
+            
+            // TODO: Might need to refactor this part
+            val itemList = (data.data["items"] as List<*>?)?.map {
```

Comment:
👍 yep, this would be a good way to parse the data inside the viewholder itself

break it out to a method to make it even easier to read what is parsing logic and what is rendering logic

Reply:
Fixed in https://github.com/ludugz/weld-rendering/pull/2/commits/d08c902b0f2959b2eb3e0fa9cb437ca0c49910c6